SimplifiedTests/NYPLAxisTests/NYPLAxisDownloadProgressTests.swift

```diff
@@ -0,0 +1,312 @@
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: encryptionURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.85/3)
+
+    // 10 % completed
+    sut.didFinishTask(with: encryptionURL)
+    XCTAssertEqual(progressListener.currentProgress, 0.1)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: encryptionURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.85/3)
+
+    // 15 % completed
+    sut.didFinishTask(with: containerURL)
+    XCTAssertEqual(progressListener.currentProgress, 0.15)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: encryptionURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.85/3)
+
+    // ~43% completed
+    sut.didFinishTask(with: chapter2URL)
+    XCTAssertEqual(progressListener.currentProgress, (0.15 + (0.85/3)).roundedToTwoDecimalPlaces)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: containerURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: encryptionURL), 0.05)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter2URL), 0.85/3)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.85/3)
+  }
+
+  func testAddingMoreTasksShouldNotAffectWeightOfCompletedTasks() {
+    sut.addFixedWeightTask(with: licenseURL, weight: 0.2)
+    sut.addFlexibleWeightTask(with: chapter1URL)
+
+    XCTAssertEqual(progressListener.currentProgress, 0.0)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.2)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.8)
+
+    sut.addFlexibleWeightTask(with: chapter3URL)
+    XCTAssertEqual(progressListener.currentProgress, 0.0)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: licenseURL), 0.2)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter1URL), 0.4)
+    XCTAssertEqual(sut.test_getWeightForProcess(withURL: chapter3URL), 0.4)
+
+    // 20 % completed
+    sut.didFinishTask(with: licenseURL)
```

Comment:
This test is named `AddingMoreTasksShouldNotAffectWeightOfCompletedTasks` but there is no task added after this task is completed. Also `licenseURL` is a fixed weight task, so it should not be affected in any ways?

Reply:
Great catch. Fixed.